src/per/i2c.cpp

```diff
@@ -371,46 +478,128 @@ I2CHandle::Impl::StartDmaTransfer(uint16_t                       address,
     while(HAL_I2C_GetState(&i2c_hal_handle_) != HAL_I2C_STATE_READY) {};
 
     // reinit the DMA
-    i2c_dma_tx_handle_.Instance = DMA1_Stream6;
+    i2c_dma_tc_handle_.Instance = DMA1_Stream6;
     switch(config_.periph)
     {
         case I2CHandle::Config::Peripheral::I2C_1:
-            i2c_dma_tx_handle_.Init.Request = DMA_REQUEST_I2C1_TX;
+            i2c_dma_tc_handle_.Init.Request = DMA_REQUEST_I2C1_TX;
             break;
         case I2CHandle::Config::Peripheral::I2C_2:
-            i2c_dma_tx_handle_.Init.Request = DMA_REQUEST_I2C2_TX;
+            i2c_dma_tc_handle_.Init.Request = DMA_REQUEST_I2C2_TX;
             break;
         case I2CHandle::Config::Peripheral::I2C_3:
-            i2c_dma_tx_handle_.Init.Request = DMA_REQUEST_I2C3_TX;
+            i2c_dma_tc_handle_.Init.Request = DMA_REQUEST_I2C3_TX;
             break;
         // I2C4 has no DMA yet. TODO
         default: return I2CHandle::Result::ERR;
     }
-    i2c_dma_tx_handle_.Init.Direction           = DMA_MEMORY_TO_PERIPH;
-    i2c_dma_tx_handle_.Init.PeriphInc           = DMA_PINC_DISABLE;
-    i2c_dma_tx_handle_.Init.MemInc              = DMA_MINC_ENABLE;
-    i2c_dma_tx_handle_.Init.PeriphDataAlignment = DMA_PDATAALIGN_BYTE;
-    i2c_dma_tx_handle_.Init.MemDataAlignment    = DMA_MDATAALIGN_BYTE;
-    i2c_dma_tx_handle_.Init.Mode                = DMA_NORMAL;
-    i2c_dma_tx_handle_.Init.Priority            = DMA_PRIORITY_LOW;
-    i2c_dma_tx_handle_.Init.FIFOMode            = DMA_FIFOMODE_DISABLE;
-    i2c_dma_tx_handle_.Init.MemBurst            = DMA_MBURST_SINGLE;
-    i2c_dma_tx_handle_.Init.PeriphBurst         = DMA_PBURST_SINGLE;
-
-    if(HAL_DMA_Init(&i2c_dma_tx_handle_) != HAL_OK)
+    i2c_dma_tc_handle_.Init.Direction           = DMA_MEMORY_TO_PERIPH;
+    i2c_dma_tc_handle_.Init.PeriphInc           = DMA_PINC_DISABLE;
+    i2c_dma_tc_handle_.Init.MemInc              = DMA_MINC_ENABLE;
+    i2c_dma_tc_handle_.Init.PeriphDataAlignment = DMA_PDATAALIGN_BYTE;
+    i2c_dma_tc_handle_.Init.MemDataAlignment    = DMA_MDATAALIGN_BYTE;
+    i2c_dma_tc_handle_.Init.Mode                = DMA_NORMAL;
+    i2c_dma_tc_handle_.Init.Priority            = DMA_PRIORITY_LOW;
+    i2c_dma_tc_handle_.Init.FIFOMode            = DMA_FIFOMODE_DISABLE;
+    i2c_dma_tc_handle_.Init.MemBurst            = DMA_MBURST_SINGLE;
+    i2c_dma_tc_handle_.Init.PeriphBurst         = DMA_PBURST_SINGLE;
+
+    if(HAL_DMA_Init(&i2c_dma_tc_handle_) != HAL_OK)
     {
         Error_Handler();
     }
 
-    __HAL_LINKDMA(&i2c_hal_handle_, hdmatx, i2c_dma_tx_handle_);
+    __HAL_LINKDMA(&i2c_hal_handle_, hdmatx, i2c_dma_tc_handle_);
 
     // start the transfer
     // TODO: Add ScopedIrqBlocker
```

Comment:
Could you secure this block (and all other lines that access these variables that are shared between a IRQ and the "main thread") with a `ScopedIrqBlocker` and remove this comment? 
I think when this code was written, we didn't yet have the `ScopedIrqBlocker`, hence the comment.